auslib/db.py

```diff
@@ -1102,6 +1102,20 @@ def validate(self, base_columns, condition_columns, changed_by, sc_id=None, tran
         self.conditions.validate(condition_columns)
         self._checkBaseTablePermissions(base_table_where, base_columns, changed_by, transaction)
 
+    def auto_signoff(self, changed_by, transaction, sc_id, dryrun, columns):
+        # - If the User scheduling a change only holds one Role, record a signoff with it.
+        # - If the User scheduling a change holds more than one Role, we cannot a Signoff, because
+        #   we don't know which Role we'd want to signoff with. The user will need to signoff
+        #   manually in these cases.
+        user_roles = self.db.getUserRoles(username=changed_by, transaction=transaction)
+        required_roles = set()
+        required_signoffs = self.baseTable.getPotentialRequiredSignoffs([columns], transaction=transaction)
+        if required_signoffs:
+            required_roles.update([rs["role"] for rs in required_signoffs])
+        if len(user_roles) == 1 and user_roles[0]["role"] in required_roles:
```

Comment:
You can consider prevent one hit in database if you check `user_roles` is equals 1 before call `getPotentialRequiredSignoffs`